src/transports/ssh.c

```diff
@@ -362,10 +362,13 @@ static int _git_ssh_authenticate_session(
 		case GIT_CREDENTIAL_SSH_KEY: {
 			git_credential_ssh_key *c = (git_credential_ssh_key *)cred;
 
-			if (c->privatekey)
+			if (c->privatekey) {
 				rc = libssh2_userauth_publickey_fromfile(
 					session, c->username, c->publickey,
 					c->privatekey, c->passphrase);
+				if (rc == LIBSSH2_ERROR_FILE)
+					rc == LIBSSH2_ERROR_AUTHENTICATION_FAILED;
```

Comment:
That's a comparison, not an assignment. Additionally, I don't understand what this fix is about, as a `LIBSSH2_ERROR_FILE` error would have been caught by the code at line 435, and wrapped into one of our own.

Reply:
Hmmm... Yes, you are right. 
Should be closed, I guess.
However, it returns `GIT_EAUTH` in this PR.